### ISSUE TITLE
Field in ValueChecker cannot be refined.

### ISSUE THREAD

**Issue by xingweitian** [typetools/checker-framework#3105]
Consider these two cases:

```java
import org.checkerframework.common.value.qual.StringVal;

class C {
    public static final String DEMO = "foo";
}

class Demo {  
  @StringVal("foo") String m() {
      return C.DEMO;
  }
}
```

```java
import org.checkerframework.common.value.qual.StringVal;

class C {
    public static final String DEMO = "foo";
}

class Demo extends C{  
  @StringVal("foo") String m() {
      return DEMO; // error
  }
}
```

The first one can pass the typechecking while the second one can't (error at `return DEMO;`):

```
Error: [return.type.incompatible] incompatible types in return.
   type of expression: @UnknownVal String
   method return type: @StringVal("foo") String
```

We should consider why the field in this case cannot be refined.

**Comment by mernst** [comment]
`DEMO` is parsed as an `IdentifierTree`, but `C.DEMO` is parsed as a `MemberSelectTree`.  I believe the bug is that ValueTreeAnnotator lacks a `visitIdentifier` method that would have similar logic to `visitMemberSelect`.  (Maybe some logic can be reused for both cases.)
